src/neo/Helper.cs

```diff
@@ -310,5 +310,88 @@ internal static BigInteger WeightedAverage<T>(this IEnumerable<T> source, Func<T
                 yield return resultSelector(item, weight);
             }
         }
+
+        public static byte[] Concat(this byte[] a, byte[] b)
+        {
+            var result = new byte[a.Length + b.Length];
+            a.CopyTo(result, 0);
+            b.CopyTo(result, a.Length);
+            return result;
+        }
+
+        public static byte[] CommonPrefix(this byte[] a, byte[] b)
+        {
+            var prefix = Array.Empty<byte>();
+            var minLen = a.Length <= b.Length ? a.Length : b.Length;
+
+            if (a.Length != 0 && b.Length != 0)
+            {
+                for (int i = 0; i < minLen; i++)
+                {
+                    if (a[i] != b[i]) break;
+                    prefix = prefix.Add(a[i]);
+                }
+            }
+            return prefix;
+        }
+
+        public static bool Equal(this byte[] a, byte[] b)
+        {
+            if (a is null || b is null) return false;
+            var len = a.Length;
+            if (len != b.Length) return false;
+            unsafe
+            {
+                fixed (byte* ap = a, bp = b)
+                {
+                    long* alp = (long*)ap, blp = (long*)bp;
+                    for (; len >= 8; len -= 8)
+                    {
+                        if (*alp != *blp) return false;
+                        alp++;
+                        blp++;
+                    }
+                    byte* abp = (byte*)alp, bbp = (byte*)blp;
+                    for (; len > 0; len--)
+                    {
+                        if (*abp != *bbp) return false;
+                        abp++;
+                        bbp++;
+                    }
+                }
+            }
+            return true;
+        }
+
+        public static byte[] Skip(this byte[] a, int count)
+        {
+            var result = Array.Empty<byte>();
+            var len = a.Length - count;
+            if (0 < len)
+            {
+                result = new byte[len];
+                Array.Copy(a, count, result, 0, len);
+            }
+            return result;
+        }
+
+        public static byte[] Add(this byte[] a, byte b)
+        {
+            var len = a.Length;
+            Array.Resize<byte>(ref a, len + 1);
+            a[len] = b;
+            return a;
+        }
+
+        public static byte[] ToNibbles(this byte[] path)
+        {
+            var result = new byte[path.Length * 2];
+            for (int i = 0; i < path.Length; i++)
+            {
+                result[i * 2] = (byte)(path[i] >> 4);
+                result[i * 2 + 1] = (byte)(path[i] & 0x0F);
```

Comment:
Is there a performance improvement in the use of the bit operators?
Don't get me wrong, I love them but sometimes they make the code harder to read and maintain with no big perf improvement, in lower-level languages they are very good but in higher-level ones I don't know for sure.

In that case, the operations could be:
```cs
result[i * 2] = (byte)(path[i] / 16);
result[i * 2 + 1] = (byte)(path[i] % 16);
```

Reply:
`>>` is faster than `/`.  `&` make no much difference with `%`.